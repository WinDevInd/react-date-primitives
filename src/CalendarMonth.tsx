import * as React from 'react';

import {
    startOfMonth,
    isDayAfter,
    isDayBefore,
    isSameDay,
    setMonth,
    getDatesofMonth
} from './utils';

export interface CalendarMonthRenderProps {
    days: Array<Array<DayOfMonth | null>>;
    month: Date;
    dropdownMonths?: CalendarDropdownProps[];
    dropdownYears?: CalendarDropdownProps[];
}

export interface CalendarDropdownProps {
    value: number;
    selected: boolean;
    disabled: boolean;
}

export interface DayOfMonth {
    date: Date;
    inRange: boolean;
    selected: boolean;
    disabled: boolean;
}

export interface CalendarMonthProps {
    /**
     * Current Month
     */
    month: Date;
    /**
     * The start of the initially selected date range
     */
    startDate?: Date;
    /**
     * The end of the initially selected date range
     */
    endDate?: Date;
    /**
     * The earliest date a user may select
     */
    minDate?: Date;
    /**
     * The latest date a user may select
     */
    maxDate?: Date;
    /**
     * If set as `true`, shows **month** and **year** dropdowns above the calendar
     */
    showDropdowns?: boolean;
    /**
     *
     */
    render(props: CalendarMonthRenderProps): React.ReactNode;
}

/**
 * A Calendar Month
 *
 * @see examples/SimpleDatePicker.tsx
 */
export class CalendarMonth extends React.Component<CalendarMonthProps> {
    private dropdownMonths?: CalendarDropdownProps[];
    private dropdownYears?: CalendarDropdownProps[];

    constructor(props: CalendarMonthProps) {
        super(props);

        if (props.showDropdowns) {
            const [dropdownMonths, dropdownYears] = this.getDropDowns(props);
            this.dropdownMonths = dropdownMonths;
            this.dropdownYears = dropdownYears;
        }
    }

    componentWillReceiveProps(nextProps: CalendarMonthProps) {
        if (nextProps.showDropdowns) {
            const [dropdownMonths, dropdownYears] = this.getDropDowns(nextProps);
            this.dropdownMonths = dropdownMonths;
            this.dropdownYears = dropdownYears;
        } else {
            this.dropdownMonths = undefined;
            this.dropdownYears = undefined;
        }
    }

    private getDaysofMonth(props: CalendarMonthProps): Array<Array<DayOfMonth | null>> {
        const { month, startDate, endDate, minDate, maxDate } = props;
        const currentMonth = getDatesofMonth(month);

        const startDateDisabled = startDate && minDate && !isDayAfter(startDate, minDate);

        return currentMonth.map((week) =>
            week.map((day) => {
                if (!day) return null;

                const disabled = Boolean(
                    (minDate && isDayBefore(day, minDate)) || (maxDate && isDayAfter(day, maxDate))
                );

                return {
                    date: day,
                    inRange:
                        !startDateDisabled &&
                        !disabled &&
                        Boolean(
                            startDate &&
                                isDayAfter(day, startDate) &&
                                endDate &&
                                isDayBefore(day, endDate)
                        ),
                    selected:
                        !disabled &&
                        Boolean(
                            (startDate && isSameDay(day, startDate)) ||
                                (endDate && isSameDay(day, endDate))
                        ),
                    disabled
                };
            })
        );
    }

    getDropDowns = (
        props: CalendarMonthProps
    ): [CalendarDropdownProps[], CalendarDropdownProps[]] => {
        const { minDate, maxDate, month } = props;

        const firstDay = startOfMonth(month);
        const today = new Date();
        const minYear = minDate ? minDate.getFullYear() : today.getFullYear() - 50;
        const maxYear = maxDate ? maxDate.getFullYear() + 1 : today.getFullYear() + 5;

        const dropdownMonths: CalendarDropdownProps[] = Array.from({ length: 12 }, (_, i) => ({
            disabled: Boolean(
                (minDate && isDayBefore(setMonth(month, i), minDate)) ||
                    (maxDate && isDayAfter(setMonth(month, i), maxDate))
            ),
            value: i,
            selected: firstDay.getMonth() === i
        }));

        const dropdownYears: CalendarDropdownProps[] = Array.from(
            { length: maxYear - minYear },
            (_, i) => ({
                value: i + minYear,
                disabled: false,
                selected: i + minYear === firstDay.getFullYear()
            })
        );

        return [dropdownMonths, dropdownYears];
    };

    render() {
        const { month, render } = this.props;

        return render({
            days: this.getDaysofMonth(this.props),
            month: startOfMonth(month),
            dropdownMonths: this.dropdownMonths,
            dropdownYears: this.dropdownYears
        });
    }
}
