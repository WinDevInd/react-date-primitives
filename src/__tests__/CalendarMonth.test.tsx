import * as React from 'react';
import { shallow } from 'enzyme';

import { CalendarMonth, CalendarMonthRenderProps } from '../CalendarMonth';
import { addDays, addMonths } from '../utils';

describe('<CalenderMonth /> tests', () => {
    const month = new Date(Date.UTC(2017, 0 /* Jan */, 1, 0, 0, 0, 0));

    test('calls render props with given month and its days', () => {
        const render = jest.fn();
        shallow(<CalendarMonth month={month} render={render} />);

        expect(render).toHaveBeenCalledTimes(1);
        const args: CalendarMonthRenderProps = render.mock.calls[0][0];
        expect(args.days).toMatchSnapshot();
        expect(args.dropdownMonths).not.toBeDefined();
        expect(args.dropdownYears).not.toBeDefined();
    });

    test('render prop has proper days selected when given', () => {
        const render = jest.fn();
        shallow(
            <CalendarMonth
                month={month}
                minDate={addDays(month, 5)}
                startDate={addDays(month, 10)}
                endDate={addDays(month, 20)}
                maxDate={addDays(month, 25)}
                render={render}
            />
        );

        const args: CalendarMonthRenderProps = render.mock.calls[0][0];
        expect(args.days).toMatchSnapshot();
    });

    test('minDate and maxDate get priorty over startDate and endDate', () => {
        const render = jest.fn();
        shallow(
            <CalendarMonth
                month={month}
                minDate={addDays(month, 10)}
                startDate={addDays(month, 5)}
                endDate={addDays(month, 25)}
                maxDate={addDays(month, 20)}
                render={render}
            />
        );

        const args: CalendarMonthRenderProps = render.mock.calls[0][0];
        expect(args.days).toMatchSnapshot();
        expect(args.days[1][0]).toMatchObject({
            selected: false,
            inRange: false
        });
    });

    test('showDropdowns gives apt props', () => {
        const render = jest.fn();
        shallow(<CalendarMonth showDropdowns month={month} render={render} />);

        const args: CalendarMonthRenderProps = render.mock.calls[0][0];
        expect(args.dropdownMonths).toMatchSnapshot();
        expect(args.dropdownYears).toMatchSnapshot();
    });

    test('dropdowns are updated when minDate, maxDate are given', () => {
        const render = jest.fn();
        const component = shallow(<CalendarMonth showDropdowns month={month} render={render} />);

        let args: CalendarMonthRenderProps = render.mock.calls[0][0];
        expect(args.dropdownMonths).toMatchSnapshot();
        expect(args.dropdownYears).toMatchSnapshot();

        component.setProps({
            minDate: addMonths(month, 2),
            maxDate: addMonths(month, 10)
        });

        args = render.mock.calls[1][0];
        expect(args.dropdownMonths[1]).toMatchObject({
            disabled: true
        });

        expect(args.dropdownMonths[11]).toMatchObject({
            disabled: true
        });
        expect(args.dropdownYears.length).toBe(1);

        expect(args.dropdownMonths).toMatchSnapshot();
        expect(args.dropdownYears).toMatchSnapshot();
    });
});
