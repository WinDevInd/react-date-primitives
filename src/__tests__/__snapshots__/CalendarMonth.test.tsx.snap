// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalenderMonth /> tests calls render props with given month and its days 1`] = `
Array [
  Array [
    Object {
      "date": 1/1/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/2/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/3/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/4/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/5/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/6/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/7/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/8/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/9/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/10/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/11/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/12/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/13/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/14/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/15/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/16/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/17/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/18/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/19/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/20/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/21/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/22/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/23/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/24/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/25/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/26/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/27/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/28/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/29/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/30/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/31/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    null,
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
]
`;

exports[`<CalenderMonth /> tests dropdowns are updated when minDate, maxDate are given 1`] = `
Array [
  Object {
    "disabled": false,
    "selected": true,
    "value": 0,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 3,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 4,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 5,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 6,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 7,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 8,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 9,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 10,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 11,
  },
]
`;

exports[`<CalenderMonth /> tests dropdowns are updated when minDate, maxDate are given 2`] = `
Array [
  Object {
    "disabled": false,
    "selected": false,
    "value": 1968,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1969,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1970,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1971,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1972,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1973,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1974,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1975,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1976,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1977,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1978,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1979,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1980,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1981,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1982,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1983,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1984,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1985,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1986,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1987,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1988,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1989,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1990,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1991,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1992,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1993,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1994,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1995,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1996,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1997,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1998,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1999,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2000,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2001,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2002,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2003,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2004,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2005,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2006,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2007,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2008,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2009,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2010,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2011,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2012,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2013,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2014,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2015,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2016,
  },
  Object {
    "disabled": false,
    "selected": true,
    "value": 2017,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2018,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2019,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2020,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2021,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2022,
  },
]
`;

exports[`<CalenderMonth /> tests dropdowns are updated when minDate, maxDate are given 3`] = `
Array [
  Object {
    "disabled": true,
    "selected": true,
    "value": 0,
  },
  Object {
    "disabled": true,
    "selected": false,
    "value": 1,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 3,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 4,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 5,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 6,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 7,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 8,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 9,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 10,
  },
  Object {
    "disabled": true,
    "selected": false,
    "value": 11,
  },
]
`;

exports[`<CalenderMonth /> tests dropdowns are updated when minDate, maxDate are given 4`] = `
Array [
  Object {
    "disabled": false,
    "selected": true,
    "value": 2017,
  },
]
`;

exports[`<CalenderMonth /> tests minDate and maxDate get priorty over startDate and endDate 1`] = `
Array [
  Array [
    Object {
      "date": 1/1/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/2/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/3/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/4/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/5/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/6/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/7/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/8/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/9/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/10/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/11/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/12/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/13/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/14/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/15/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/16/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/17/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/18/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/19/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/20/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/21/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/22/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/23/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/24/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/25/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/26/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/27/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/28/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/29/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/30/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/31/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    null,
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
]
`;

exports[`<CalenderMonth /> tests render prop has proper days selected when given 1`] = `
Array [
  Array [
    Object {
      "date": 1/1/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/2/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/3/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/4/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/5/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/6/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/7/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/8/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/9/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/10/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/11/2017,
      "disabled": false,
      "inRange": false,
      "selected": true,
    },
    Object {
      "date": 1/12/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/13/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/14/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/15/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/16/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/17/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/18/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/19/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/20/2017,
      "disabled": false,
      "inRange": true,
      "selected": false,
    },
    Object {
      "date": 1/21/2017,
      "disabled": false,
      "inRange": false,
      "selected": true,
    },
  ],
  Array [
    Object {
      "date": 1/22/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/23/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/24/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/25/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/26/2017,
      "disabled": false,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/27/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/28/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
  ],
  Array [
    Object {
      "date": 1/29/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/30/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    Object {
      "date": 1/31/2017,
      "disabled": true,
      "inRange": false,
      "selected": false,
    },
    null,
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
]
`;

exports[`<CalenderMonth /> tests showDropdowns gives apt props 1`] = `
Array [
  Object {
    "disabled": false,
    "selected": true,
    "value": 0,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 3,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 4,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 5,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 6,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 7,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 8,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 9,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 10,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 11,
  },
]
`;

exports[`<CalenderMonth /> tests showDropdowns gives apt props 2`] = `
Array [
  Object {
    "disabled": false,
    "selected": false,
    "value": 1968,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1969,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1970,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1971,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1972,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1973,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1974,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1975,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1976,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1977,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1978,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1979,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1980,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1981,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1982,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1983,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1984,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1985,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1986,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1987,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1988,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1989,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1990,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1991,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1992,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1993,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1994,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1995,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1996,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1997,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1998,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 1999,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2000,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2001,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2002,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2003,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2004,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2005,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2006,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2007,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2008,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2009,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2010,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2011,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2012,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2013,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2014,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2015,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2016,
  },
  Object {
    "disabled": false,
    "selected": true,
    "value": 2017,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2018,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2019,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2020,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2021,
  },
  Object {
    "disabled": false,
    "selected": false,
    "value": 2022,
  },
]
`;
